File: c/target_program.c
// target_program.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>
#include <time.h>

void generate_random_bytes(unsigned char *buffer, size_t length) {
    for (size_t i = 0; i < length; i++) {
        buffer[i] = rand() % 256;
    }
}

int main() {
    // Initialize random seed with time and PID for better randomness
    srand(time(NULL) ^ getpid());
    
    // Generate and display our target 16 random bytes
    unsigned char target_bytes[16];
    generate_random_bytes(target_bytes, sizeof(target_bytes));
    
    printf("=== TARGET PROGRAM ===\n");
    printf("PID: %d\n", getpid());
    printf("Target 16 bytes (hex): ");
    for (int i = 0; i < 16; i++) {
        printf("%02x ", target_bytes[i]);
    }
    printf("\n");
    
    printf("Target 16 bytes (decimal): ");
    for (int i = 0; i < 16; i++) {
        printf("%d ", target_bytes[i]);
    }
    printf("\n");
    
    // Store the bytes in various memory locations to make them easier to find
    unsigned char *heap_copy = malloc(16);
    memcpy(heap_copy, target_bytes, 16);
    
    static unsigned char static_copy[16];
    memcpy(static_copy, target_bytes, 16);
    
    unsigned char stack_copy[16];
    memcpy(stack_copy, target_bytes, 16);
    
    printf("Bytes stored in:\n");
    printf("  Heap:   %p\n", (void*)heap_copy);
    printf("  Stack:  %p\n", (void*)stack_copy);
    printf("  Static: %p\n", (void*)static_copy);
    printf("\nProgram waiting for memory dump...\n");
    printf("Press Enter to exit or let memory dumper attach...\n");
    
    // Keep the program alive
    getchar();
    
    free(heap_copy);
    return 0;
}

File: target_go/target_program.go
package main

import (
	"bufio"
	"crypto/rand"
	"fmt"
	"os"
)

// generateRandomBytes generates cryptographically random bytes
func generateRandomBytes(length int) ([]byte, error) {
	buffer := make([]byte, length)
	_, err := rand.Read(buffer)
	if err != nil {
		return nil, err
	}
	return buffer, nil
}

func main() {
	// Generate 16 random bytes
	targetBytes, err := generateRandomBytes(16)
	if err != nil {
		fmt.Fprintf(os.Stderr, "Error generating random bytes: %v\n", err)
		os.Exit(1)
	}

	fmt.Println("=== TARGET PROGRAM ===")
	fmt.Printf("PID: %d\n", os.Getpid())
	
	// Display bytes in hex format
	fmt.Print("Target 16 bytes (hex): ")
	for _, b := range targetBytes {
		fmt.Printf("%02x ", b)
	}
	fmt.Println()

	// Display bytes in decimal format
	fmt.Print("Target 16 bytes (decimal): ")
	for _, b := range targetBytes {
		fmt.Printf("%d ", b)
	}
	fmt.Println()

	// Store bytes in multiple locations to make them easier to find
	// Heap allocation
	heapCopy := make([]byte, 16)
	copy(heapCopy, targetBytes)

	// Another heap allocation
	staticCopy := make([]byte, 16)
	copy(staticCopy, targetBytes)

	// Stack allocation
	var stackCopy [16]byte
	copy(stackCopy[:], targetBytes)

	fmt.Println("Bytes stored in:")
	fmt.Printf("  Heap (slice 1): %p\n", heapCopy)
	fmt.Printf("  Heap (slice 2): %p\n", staticCopy)
	fmt.Printf("  Stack (array):  %p\n", &stackCopy)
	fmt.Println("\nProgram waiting for memory dump...")
	fmt.Println("Press Enter to exit or let memory dumper attach...")

	// Keep the program alive and prevent garbage collection
	reader := bufio.NewReader(os.Stdin)
	reader.ReadString('\n')

	// Use the variables to prevent compiler optimization
	_ = heapCopy
	_ = staticCopy
	_ = stackCopy
}
